### PR TITLE
Add text field for time votes

### PR DESCRIPTION
**Pull request by alice-blue**
Add "time_note" column to the playertimes table.

Add a text field in the time controls. Display the user's note along with the time if one is already saved.

In the list of time votes for a given game, display the notes alongside the times. In admin mode, also display the userid.

Thie userid part should address https://github.com/iftechfoundation/ifdb/issues/1036

### CODE REVIEW

**Comment by alice-blue** [comment]
I wonder if this should pull the rounded median time of a game from gametimes_mv instead of calculating it from the individual player times?

File: www/viewgame [hunk 1]
```diff
@@ -1456,7 +1464,8 @@ function submitTime() {
         return;
     }
     // Time is valid, as far as we can tell
-    jsonSend("settime?game=<?php echo $id?>&newtime=" + new_time, "timeSaveMsg", swapTimeSections);
+    var time_note = document.getElementById('timeNote').value;
+    jsonSend("settime?game=<?php echo $id?>&newtime=" + new_time + "&note=" + time_note, "timeSaveMsg", swapTimeSections);
```
Comment (salty-horse) [inline www/viewgame:1468]: The time note needs to be URL-encoded with `encodeURIComponent(time_note)`.

And you can use a javascript template literal:
```javascript
`settime?game=<?php echo $id?>&newtime=${new_time}&note=${encodeURIComponent(time_note})`
```

File: www/viewgame [hunk 2]
```diff
@@ -1970,27 +1983,36 @@ if ($mediantime > 60) {
...
-    foreach ($alltimes as $t) {
-        $time_vote_text = convertTimeToText($t);
-        echo "<li>$time_vote_text</li>";
+    foreach ($times_with_notes as $t) {
+        $time_in_text = convertTimeToText($t[time_in_minutes]);
+        $note = $t[time_note];
+        $voter = $t[userid];
+        $list_item = "<strong>$time_in_text</strong>";
+        if ($note != "") {
+            $list_item .= "—\"$note\"";      // If the voter added a note, display it
```
Comment (salty-horse) [inline www/viewgame:2002]: Need to protect from HTML injection. @dfabulich might know how.

**Review by salty-horse — CHANGES_REQUESTED** [review]

File: www/viewgame [hunk 3]
```diff
@@ -1367,19 +1367,24 @@ function updateUnwishList(id, stat)
...
     $mystoredtime=0;
+    $mystoredtimenote="";
 
-    $result = mysqli_execute_query($db, "select time_in_minutes from playertimes where gameid = ? and userid = ?", [$id, $curuser]);
+    $result = mysqli_execute_query($db, "select time_in_minutes, time_note from playertimes where gameid = ? and userid = ?", [$id, $curuser]);
     if (!$result) throw new Exception("Error: " . mysqli_error($db));
         while ($row = mysqli_fetch_assoc($result)) {     
             // There should only be one row, at most
             $mystoredtime = $row['time_in_minutes']; 
+            $mystoredtimenote = $row['time_note'];
```
[the remark this answers is not in the document]
Comment (alice-blue) [reply www/viewgame]: Changed this

File: www/viewgame [hunk 4]
```diff
@@ -1367,19 +1367,24 @@ function updateUnwishList(id, stat)
...
         }
     // At this point $mystoredtime should reflect the player's actual stored time for this game
     $my_time_as_text = "";
     $time_announcement = ""; // The string shown in the Estimated Play Time control panel
     if ($mystoredtime >= 1) {
         $my_time_as_text=convertTimeToText($mystoredtime);
         $time_announcement = "Your vote: " . $my_time_as_text;
+        if ($mystoredtimenote != "") {
```
[the remark this answers is not in the document]
Comment (alice-blue) [reply www/viewgame]: Changed this

File: www/viewgame (hunk 1 above)
Comment (alice-blue) [reply www/viewgame]: Changed this

File: www/viewgame (hunk 2 above)
Comment (alice-blue) [reply www/viewgame:1998]: Changed this

**Comment by alice-blue** [comment]
@dfabulich since the other PR is merged, should I try to resolve the conflicts in this one?

**Comment by dfabulich** [comment]
Please do! (Or I can do it, but, later)

**Comment by alice-blue** [comment]
Ok, I think it's resolved. I didn't test it again, though.

**Comment by alice-blue** [comment]
This may need more discussion

File: www/viewgame [hunk 5]
```diff
@@ -1515,7 +1523,8 @@ function submitTime() {
         return;
     }
     // Time is valid, as far as we can tell
-    jsonSend("settime?game=<?php echo $id?>&newtime=" + new_time, "timeSaveMsg", swapTimeSections);
+    var time_note = document.getElementById('timeNote').value;
+    jsonSend("settime?game=<?php echo $id?>&newtime=${new_time}&note=${encodeURIComponent(time_note}), "timeSaveMsg", swapTimeSections);
```
Comment (dfabulich) [inline www/viewgame:1527]: This line is a syntax error. I think you mean:

```
jsonSend(`settime?game=<?= $id?>&newtime=${new_time}&note=${encodeURIComponent(time_note)}`, "timeSaveMsg", swapTimeSections);
```

**Review by dfabulich — CHANGES_REQUESTED** [review]

File: www/viewgame (hunk 5 above)
Comment (alice-blue) [reply www/viewgame]: Ok, I copy-pasted your code in there...hopefully I got it right this time.

**Comment by dfabulich** [comment]
Not sure which PR you want to work from now; we should either close this or #1153, now that we're switching to all public user names.

**Comment by alice-blue** [comment]
Ok, I'll close this one.

**Pull request closed without merging**